tests/simulator/py/fluid_init.py

```diff
@@ -0,0 +1,244 @@
+from phare.pp.diagnostic.patch import aggregate_level0_patch_domain
+
+
+class FluidInitValidation(InitValueValidation):
+    min_interp = 1
+    max_interp = 1
+
+    def pop_diag_file(fn_name):
+        return "ions_pop_ions_${POP}_" + fn_name
+
+    def test_1d(self):
+        from phare.pp.diagnostics import _FluidPatchData
+        from tests.diagnostic import dump_all_diags
+
+        dim = 1
+
+        min_interp = FluidInitValidation.min_interp
+        max_interp = FluidInitValidation.max_interp + 1
+        for interp in range(min_interp, max_interp):
+            diag_out_dir = "phare_outputs/fluid_init" + str(dim) + "_" + str(interp)
+            simInputs = InitValueValidation.diag_options(diag_out_dir)
+            simInputs.update(
+                {
+                    "diags_fn": lambda model: dump_all_diags(model.populations),
+                    "populations": {
+                        "beta": {"nbr_part_per_cell": 10000, "init": {"seed": 133337}},
+                        "gamma": {
+                            "nbr_part_per_cell": 100000,
+                            "init": {"seed": 1333337},
+                        },
+                    },
+                }
+            )
+
+            diags = self.runAndDump(dim=1, interp=interp, input=simInputs)[
+                _FluidPatchData.__name__
+            ]
+            self.checkPopulationDensityEqualsUserFunction(diags, interp)
+            self.checkPopulationFluxEqualsUserFunction(diags, interp)
+            self.checkIonDensityAndVelocityConsistentWithPopulations(diags, interp)
+
+    def checkIonDensityAndVelocityConsistentWithPopulations(self, fluid_diags, interp):
+        tolerance = 1e-6
+        sim = ph.globals.sim
+        model_dict = ph.globals.sim.model.model_dict
+
+        pop_densities, pop_fluxes = [
+            _pop_name_to_patch_level0(
+                fluid_diags, FluidInitValidation.pop_diag_file(fn_name=name)
+            )
+            for name in ["density", "flux"]
+        ]
+
+        total_density = _sum_pop_patches(pop_densities, "density")
+        ions_density_diag = selectDiag(fluid_diags, "ions_density")
+        ions_density = aggregate_level0_patch_domain(
+            ions_density_diag.levels[0], shared_patch_border=True
+        )["density"]
+
+        np.testing.assert_allclose(
+            total_density, ions_density, rtol=tolerance,
+        )
+
+        total_flux = {key: _sum_pop_patches(pop_fluxes, key) for key in ["x", "y", "z"]}
+        ions_bulkVelocity_diag = selectDiag(fluid_diags, "ions_bulkVelocity")
+        ions_bulkVelocity = aggregate_level0_patch_domain(
+            ions_bulkVelocity_diag.levels[0], shared_patch_border=True
+        )
+
+        for vel, flux in ions_bulkVelocity.items():
+            np.testing.assert_allclose(
+                total_flux[vel] / total_density, flux, rtol=tolerance
+            )
+
+    def checkPopulationFluxEqualsUserFunction(self, fluid_diags, interp):
```

Comment:
so, I think I sort of understand what the code is doing, but mostly because I'm the one who told you what to do :D if I was totally new here, or if I am me in a year from now, or you in a year too, I am not sure at all we'll get all that.
can we add comments not about code but about what we want to test and why, why we chose these 0.09 and 0.007 thresholds, etc.

Reply:
can you suggest something better than 

"we picked them because they worked?"